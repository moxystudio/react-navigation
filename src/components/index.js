export { default as Navbar } from './navbar';
export { default as Drawer } from './drawer';
