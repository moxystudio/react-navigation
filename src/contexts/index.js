export { default as NavigationContext } from './NavigationContext';
