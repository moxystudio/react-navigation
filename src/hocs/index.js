export { default as withNavigation } from './withNavigation';
