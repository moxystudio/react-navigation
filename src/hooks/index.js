export { default as useDocumentScrollThrottled } from './useDocumentScrollThrottled';
