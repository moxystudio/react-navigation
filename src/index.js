export * from './components';
export * from './hocs';
export * from './hooks';
export * from './providers';
