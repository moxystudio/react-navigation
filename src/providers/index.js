export { default as NavigationProvider } from './NavigationProvider';
