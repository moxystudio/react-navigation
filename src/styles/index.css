@import "./navbar.css";
@import "./drawer.css";
