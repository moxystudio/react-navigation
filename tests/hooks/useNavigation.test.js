import React from 'react';
import { render } from '@testing-library/react';
import stringifyObject from '../utils/stringify-object';
import { NavigationProvider, useNavigation } from '../../src';

const MyTestComponent = () => {
    const navigation = useNavigation();

    return JSON.stringify(stringifyObject(navigation));
};

describe('useNavigation Hook', () => {
    it('should provide navigation', () => {
        const { asFragment } = render(
            <NavigationProvider>
                <MyTestComponent />
            </NavigationProvider>,
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('should throw error if no NavigationProvider', () => {
        jest.spyOn(console, 'error');
        console.error.mockImplementation(() => {});

        expect(() => {
            render(<MyTestComponent />);
        }).toThrow();

        console.error.mockRestore();
    });
});
