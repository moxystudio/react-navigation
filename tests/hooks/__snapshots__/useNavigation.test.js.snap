// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useNavigation Hook should provide navigation 1`] = `
<DocumentFragment>
  {"drawer":{"isOpen":"false","toggle":"function","open":"function","close":"function"}}
</DocumentFragment>
`;
